deployment/grid/terraform/compute_plane/eks.tf

```diff
@@ -0,0 +1,383 @@
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+# Licensed under the Apache License, Version 2.0 https://aws.amazon.com/apache-2-0/
+
+
+locals {
+  chart_version = {
+    aws_for_fluentbit      = "0.1.28"
+    aws_cloudwatch_metrics = "0.0.9"
+    cluster_autoscaler     = "9.29.1"
+    keda                   = try(var.k8s_keda_version, "2.11.2")
+    influxdb               = "4.12.3"
+    prometheus             = "23.3.0"
+    grafana                = "6.58.8"
+  }
+}
+
+
+module "eks" {
+  source  = "terraform-aws-modules/eks/aws"
+  version = "~> 19.0"
+
+  cluster_name    = var.cluster_name
+  cluster_version = var.kubernetes_version
+
+  cluster_endpoint_public_access  = true
+  cluster_endpoint_private_access = var.enable_private_subnet
+  cluster_enabled_log_types       = ["api", "audit", "authenticator", "controllerManager", "scheduler"]
+
+  vpc_id     = var.vpc_id
+  subnet_ids = var.vpc_private_subnet_ids
+
+  # Node IAM Role  
+  create_iam_role = true
+
+  # EKS Managed Node Group(s)
+  eks_managed_node_group_defaults = {
+    ami_type                              = "AL2_x86_64"
+    instance_types                        = ["m6i.xlarge", "m6id.xlarge", "m6a.xlarge", "m6in.xlarge", "m5.xlarge", "m5d.xlarge", "m5a.xlarge", "m5ad.xlarge", "m5n.xlarge"]
+    attach_cluster_primary_security_group = false
+  }
+
+  eks_managed_node_groups = local.eks_worker_group_map
```

Comment:
Have you  evaluated if EKS managed node group support graceful termination ?

Reply:
As discussed in person, this is in prep for a future PR where this shouldnt be an issue.